packages/test-utils/src/reporter.ts

```diff
@@ -971,7 +995,12 @@ class ReplayReporter<TRecordingMetadata extends UnstructuredMetadata = Unstructu
     this._pendingWork.push(
       ...toUpload
         .flatMap(result => result.recordings)
-        .filter(r => (this._filter ? this._filter(r) : true))
+        .filter(r => {
+          if (this._filter && this._filter(r)) {
+            return !this._uploadedRecordings.has(r);
```

Comment:
After consideration, I don't quite like this. It probably would be better to "consume" those stored results and remove them from memory altogether before passing them to `_uploadRecording`. This way we also wouldn't try to upload the same recording twice.

I probably should recheck if such a change won't impact flakiness detection that is done in the switch/cace above

Reply:
Want me to hold off reviewing then?